src/Schema.php

```diff
@@ -397,4 +400,41 @@ private function getJsonColumns(TableSchemaInterface $table): array
 
         return $result;
     }
+
+    /**
+     * @param Check[] $checks
+     *
+     * @psalm-return list<string>|null
+     */
+    private function tryGetEnumValuesFromCheck(string $name, array $checks): ?array
+    {
+        if (empty($checks)) {
+            return null;
+        }
+
+        foreach ($checks as $check) {
+            if (!str_starts_with($check->expression, "$name ")) {
+                continue;
+            }
+
+            preg_match_all(
+                "~ IN\s*\(\s*(?:'(?:[^']+|'(?:''|[^']))*')+~i",
```

Comment:
```suggestion
                "~ IN\s*\(\s*(?:'(?:''|[^'])')~i",
```

Perhaps to check only the first value is enough?